lib/spack/spack/filesystem_view.py

```diff
@@ -690,8 +690,17 @@ def skip_list(file):
         # For compatibility, we have to create a merge_map dict mapping
         # full_src => full_dst
         files_per_spec = itertools.groupby(visitor.files.items(), key=lambda item: item[1][0])
+        prefix_to_files = dict()
+        for src_root, rel_paths in files_per_spec:
+            prefix_to_files[src_root] = list(rel_paths)
```

Comment:
If I replace the prior 3 lines with:

```
prefix_to_files = dict(files_per_spec)
```

then nothing actually gets linked into the view. I'm not clear why, but this is more verbose than one might opt for on account of that.